mirrord/layer/src/file/hooks.rs

```diff
@@ -478,6 +478,25 @@ unsafe extern "C" fn stat_detour(raw_path: *const c_char, out_stat: *mut stat) -
     result
 }
 
+/// Hook for libc's stat syscall wrapper.
+#[hook_guard_fn]
+pub(crate) unsafe extern "C" fn __xstat_detour(
+    ver: c_int,
+    raw_path: *const c_char,
+    out_stat: *mut stat,
+) -> c_int {
+    let path = (!raw_path.is_null()).then(|| CStr::from_ptr(raw_path));
+    let (Ok(result) | Err(result)) = xstat(Some(ver), Some(path), None, true)
```

Comment:
I think the version check should be done here as it's not shared logic..
also, according to docs it should be `3`
https://refspecs.linuxfoundation.org/LSB_1.1.0/gLSB/baselib-xstat-1.html
I assume you saw version 1 being used?

Reply:
yeah when I traced it, I saw version 1 being used

Reply:
verdict on this? I only saw `1` being used in my runs, do we allow both?

disassembly of the function:
```
int __xstat(int __ver,char *__filename,stat64 *__stat_buf)

{
  undefined4 *in_FS_OFFSET;
  
  if ((uint)__ver < 2) {
    syscall();
    return 4;
  }
  *in_FS_OFFSET = 0x16;
  return -1;
}
```

Reply:
sounds okay